File: client/src/api/themes.ts
import { API_BASE_URL } from "../config";

export interface Theme {
  id: string;
  name: string;
  slug: string;
}

export interface ThemeCategory {
  id: string;
  name: string;
  slug: string;
  children: Theme[];
}

export interface ThemesResponse {
  categories: ThemeCategory[];
}

export const fetchThemes = async (): Promise<ThemesResponse> => {
  const response = await fetch(`${API_BASE_URL}/themes`);
  if (!response.ok) {
    throw new Error("Failed to fetch themes");
  }
  return response.json();
};

File: server/the_history_atlas/api/types/themes.py
from uuid import UUID

from pydantic import BaseModel


class ThemeResponse(BaseModel):
    id: UUID
    name: str
    slug: str


class ThemeCategoryResponse(BaseModel):
    id: UUID
    name: str
    slug: str
    children: list[ThemeResponse]


class ThemesResponse(BaseModel):
    categories: list[ThemeCategoryResponse]
